pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -616,7 +616,7 @@ def __resolve_ckpt_dir(self, trainer: "pl.Trainer") -> None:
 
         self.dirpath = ckpt_path
 
-        if not trainer.fast_dev_run and trainer.should_rank_save_checkpoint:
+        if not trainer.fast_dev_run and trainer.training_type_plugin.should_rank_save_checkpoint:
```

Comment:
Do we still need this makedirs call here at all? Should this be part of the IO plug-in when saving?

Reply:
Yeah, I don't think it's required. Updated!

Reply:
i was hoping to remove this too but I think it may require adding makedirs to the TTPs that  do not call IO plugin's save: https://app.circleci.com/pipelines/github/PyTorchLightning/pytorch-lightning/43772/workflows/96d3a810-fd24-4edf-98c0-bee485122df7/jobs/130419,  same as #9615 previous commit tests